sources/osg/Uniform.js

```diff
@@ -54,15 +54,25 @@ Uniform.prototype = {
 
     apply: function UniformApply( gl, location ) {
 
-        if ( !this._cache )
+        if ( !this._cache || this._cacheDirty === true ) {
             this._cache = gl[ this._glCall ];
-
+            this._dirtyCache = false;
```

Comment:
> this._cacheDirty = false;
arg, well spotted!
> Also why exposing gl call signature? I think it's supposed to be kept private
I guess I did it for convenience so each time glSignature changes it dirties the cache.
https://github.com/jtorresfabra/osgjs/blob/653a94d5a8f160be5fa6aa2756ccc543325ec33b/sources/osg/Uniform.js#L169
 But I can keep it private. I'll fix it.